### ISSUE TITLE
Bad inference

### ISSUE THREAD

**Issue by keyssson** [alexiglad/EBT#9]
Hello, 
    So I'm trying to train a EBT model myself and with the instruction I successfully got one System-1-Thinking-EBT. But when I directly used it to do some inference job(like inference on GSM8k), the result was a disaster. The accuracy was nearly zero and the generation was a mess.

<img width="865" height="235" alt="Image" src="https://github.com/user-attachments/assets/3fa72815-cb47-4145-854d-92d0349a6c3d" />

<img width="474" height="331" alt="Image" src="https://github.com/user-attachments/assets/366c80ec-425e-4ce3-b248-824eeb68aa15" />
    
Is that something wrong with my training? I early stopped it at "epoch 179". Or do I need to do some fine-tune? Please enlighten me.

<img width="1865" height="796" alt="Image" src="https://github.com/user-attachments/assets/d8859411-bc92-4749-8ea8-2ccb4d556ab9" />

**Comment by alexiglad** [comment]
I'm a bit confused by what you mean when you say "I early stopped it at "epoch 179"", that seems impossible if you're training on a large pretraining dataset like pajama to have trained for 179 epochs? Can you explain what you did more?

Can you try the `baseline_transformer` model to see if it's an issue with your setup? Maybe test it when it is getting around the same pretraining perplexity.

**Comment by alexiglad** [comment]
My guess is you were training on too small of a dataset (sample, instead of e.g. sample 100B).

I'm a bit confused, as training on 8XA100s should definitely have performed more training steps, maybe check for high GPU utilization?

Another thing you can do is train the baseline to see if you get similar poor performance, or train larger models since it seems you have the resources for it. But my current guess is the pretraining dataset being sample was too small and the model overfit.

**Comment by alexiglad** [comment]
Closing this issue as there has been no response for 2 weeks but let me know if there are any other issues!
